### PR TITLE
Fixing potential bug

### PR DESCRIPTION
**Pull request by rakatyal**

### CODE REVIEW

**Comment by JKillian** [comment]
Hi @rakatyal, thanks for the PR! Is there a corresponding issue for this? Could you add a test case that would be broken before your PR and is now working correctly?

File: src/rules/adjacentOverloadSignaturesRule.ts [hunk 1]
```diff
@@ -58,7 +58,7 @@ class AdjacentOverloadSignaturesWalker extends Lint.RuleWalker {
             if (member.name !== undefined) {
                 const methodName = getTextOfPropertyName(member.name);
                 if (methodName !== undefined) {
-                    if (seen[methodName] && last !== methodName) {
+                    if (hasProperty(seen, methodName) && last !== methodName) {
```
Comment (adidahiya) [inline src/rules/adjacentOverloadSignaturesRule.ts]: why not just make this `seen.hasOwnProperty(methodName)`?

**Comment by rakatyal** [comment]
@JKillian: There isn't any corresponding issue. I have added the corresponding test case.

**Comment by adidahiya** [comment]
Thanks for revising, but the rationale behind this fix is still not immediately clear from the diff -- can you explain it in words at least, if not in a code comment?

**Comment by rakatyal** [comment]
I think when looping over the members of the node, the parents' object properties are also included. The rule checks if the given method name has been encountered and due to this when someone tries to override known methods like toString(), toLowerCase() which are defined in parents' objects, it fails even if they are only defined once.
Hence we have to use hasOwnProperty instead of just using seen[methodName].

**Pull request merged**